### ISSUE TITLE
"#" can't be use in a password

### ISSUE THREAD

**Issue by riolu452** [wrouesnel/postgres_exporter#306]
I am using Linux CentOS 7. The database I'm trying to connect to has a # in its password.
For example the DATA_SOURCE_NAME will look as follow: 

DATA_SOURCE_NAME="postgresql://user:pass#word@IP:Port/database"

I will get the following if I try to connect:

Established new database connection to "user:pass"
Error opening connection to database (postgresql://user:pass#word@IP:Port/database): dial tcp: address tcp/pass: unknown port  source="postgres_exporter.go:1319"

In other words everything after the # (# included) is removed from the data source name.
Is there a way is parse the # into a string?

**Comment by priithaamer** [comment]
Also, having `=` in password is causing `Error opening connection to database (postgresql://user:password=@host/db): malformed dsn \"dbname=db host=host password=password= user=user\""`
This line seems to be the culprit: https://github.com/wrouesnel/postgres_exporter/blob/master/cmd/postgres_exporter/postgres_exporter.go#L709

**Comment by wrouesnel** [comment]
Not easy to resolve since it depends on how to get the credentials to the underlying postgres driver. I suspect it means we need to move away from constructing the URLs.

**Comment by f4t** [comment]
Have you tried using "Percent-encoding" in your URI ?
https://www.postgresql.org/docs/11/libpq-connect.html#id-1.7.3.8.3.6

`Percent-encoding may be used to include symbols with special meaning in any of the URI parts, e.g. replace = with %3D.`

**Comment by sysadmind** [comment]
Thanks for the report! This issue has been inactive for a long time and I believe that this should be resolved already. If this is still a problem with v0.20.0, please open a new issue with up-to-date details and we'll take a look. Closing as stale for now.
